x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/SetEnabledRequest.java

```diff
@@ -40,9 +40,13 @@ public SetEnabledRequest(StreamInput in) throws IOException {
     @Override
     public ActionRequestValidationException validate() {
         ActionRequestValidationException validationException = null;
-        Error error = Users.validateUsername(username, true, Settings.EMPTY);
-        if (error != null) {
-            validationException = addValidationError(error.toString(), validationException);
+        if (username == null) {
+            validationException = addValidationError("username must be set", validationException);
+        } else {
+            Error error = Validation.Users.validateUsername(username, true, Settings.EMPTY);
+            if (error != null) {
+                validationException = addValidationError(error.toString(), validationException);
+            }
         }
```

Comment:
Good question. I had same thought when I was checking `PutUserRequest` and `TransportPutUserAction`. I was confused that `null` check and validation were split. I thought that there was a good reason for this. Now that I look at it again, I don't see a reason why not. It would make `PutUserRequest` and `SetEnabledRequest` more consistent.